src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -253,52 +253,97 @@ class EntityClient (requestContext: RequestContext) extends Actor {
     EntityUpdateDefinition(row.headOption.get,entityType,ops ++ collectionMemberAttrOp )
   }
 
-  def addToCollectionTypeFromTSV( entityType: String, memberType: String, membersAttributeName: String, tsv: TSVLoadFile ) = {
-    // group together updates to distinct entities
-    tsv.tsvData groupBy(_(0)) map { case (entityName, rows) =>
-      val ops = rows map { row =>
-        val attrRef = AttributeReference(memberType,row(1)) //row(1) is the entity to add to the entity in row.head
-        Map(addListMemberOperation,"attributeListName"->AttributeString(membersAttributeName),"newMember"->attrRef)
-      }
-      EntityUpdateDefinition(entityName,entityType,ops)
+  private def validateMembershipTSV(tsv: TSVLoadFile, membersType: Option[String]) (op: => Unit): Unit = {
+    //This magical list of conditions determines whether the TSV is populating the "members" attribute of a collection type entity.
```

Comment:
I will fix this, but meantime this will change your life for the better:
https://github.com/mdo/github-wide